### PR TITLE
Fixes azure compute plugin

### PR DESCRIPTION
**Pull request by wassoares**
## Additional information
 - This branch depends on a branch of the same name in the Fogbow Common Project.

### Note:
 - Some tests must be corrected before doing the merge.

### CODE REVIEW

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java [hunk 1]
```diff
@@ -138,19 +147,31 @@ public ComputeInstance getInstance(ComputeOrder computeOrder, AzureUser azureUse
...
         return buildComputeInstance(azureGetVirtualMachineRef, azureUser);
     }
 
+    @VisibleForTesting
+    void updateInstanceAllocation(ComputeOrder computeOrder, VirtualMachineSize virtualMachineSize) {
+        synchronized (computeOrder) {
+            int vCPU = virtualMachineSize.numberOfCores();
+            int memory = virtualMachineSize.memoryInMB();
+            int disk = computeOrder.getDisk();
+            int instances = 1;
```
Comment (jadsonluan) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java]: Remove the magic number and add a constant

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureVirtualMachineOperation.java [hunk 2]
```diff
@@ -1,28 +1,34 @@
...
+            int vCpuRequired,
+            @NotBlank String regionName, 
+            @NotNull AzureUser azureCloudUser) throws FogbowException;
 
-    AzureGetVirtualMachineRef doGetInstance(String azureInstanceId, AzureUser azureUser)
-            throws UnauthenticatedUserException, UnexpectedException,
-            NoAvailableResourcesException, InstanceNotFoundException;
+    AzureGetVirtualMachineRef doGetInstance(
+            @NotBlank String azureInstanceId, 
+            @NotNull AzureUser azureUser) throws FogbowException;
```
Comment (jadsonluan) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureVirtualMachineOperation.java]: Standartize the AzureUser parameter's name.

**Suggestion:** change all parameters named azureCloudUser to azureUser

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java (hunk 1 above)
Comment (wassoares) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java]: When this branch merges with develop this variable will no longer exist.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java [hunk 3]
```diff
@@ -55,25 +68,24 @@ public String requestInstance(ComputeOrder computeOrder, AzureUser azureUser) th
...
+        String osComputeName = computeOrder.getName();
 
         AzureCreateVirtualMachineRef azureCreateVirtualMachineRef = AzureCreateVirtualMachineRef.builder()
                 .virtualMachineName(virtualMachineName)
-                .azureGetImageRef(azureVirtualMachineImage)
+                .azureGetImageRef(imageRef)
                 .networkInterfaceId(networkInterfaceId)
                 .diskSize(diskSize)
-                .size(virtualMachineSizeName)
+                .size(virtualMachineSize.name())
```
Comment (fgan1) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java]: Extract it to a variable in order to follow the pattern.

```suggestion
String size = virtualMachineSize.name();
...
                .size(size)
...
```

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java [hunk 4]
```diff
@@ -91,42 +103,40 @@ String getUserData(ComputeOrder computeOrder) {
...
-                    .configure(azureCloudUser)
-                    .resourceGroupName(this.defaultResourceGroupName)
-                    .resourceName(this.defaultNetworkInterfaceName)
-                    .structure(AzureIdBuilder.NETWORK_INTERFACE_STRUCTURE)
-                    .build();
+            return AzureResourceIdBuilder.configure(AzureConstants.NETWORK_INTERFACE_STRUCTURE)
+                    .withSubscriptionId(azureCloudUser.getSubscriptionId())
+                    .withResourceGroupName(this.defaultResourceGroupName)
+                    .withResourceName(this.defaultNetworkInterfaceName)
+                    .buildResourceId();
```
Comment (fgan1) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java]: What do you think in use just "build()" as the method name? The "Azure ResourceId Builder" name already imply that It'll build a "resourceId".

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureResourceIdBuilder.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        public String buildResourceId() {
+            return String.format(this.structure, this.subscriptionId, this.resourceGroupName, this.resourceName);
+        }
+
+        /**
+         * It checks the resource name size in relation to Database Instance Order Id Maximum Size.
+         * It happens because the resource name makes up the instance Id. Also, it might happen due
+         * to the fact that the user choose some values such as resourceName and resourceGroupName.
+         */
+        public void checkIdSizePolicy() throws InvalidParameterException {
```
Comment (fgan1) [inline src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureResourceIdBuilder.java]: Is it still necessary ? Now the resource name is constant. let's discuss about it tomorrow.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/sdk/AzureVirtualMachineOperationSDKTest.java [hunk 6]
```diff
@@ -221,8 +217,7 @@ public void testFindVirtualMachineSizeByNameFail()
     // test case: When calling the findVirtualMachineSizeName method with two virtual machine size name that
     // fits in the requirements, it must verify if It returns the smaller virtual machine size.
     @Test
-    public void testFindVirtualMachineSizeNameSuccessfully()
-            throws NoAvailableResourcesException, UnauthenticatedUserException, UnexpectedException {
+    public void testFindVirtualMachineSizeNameSuccessfully() throws FogbowException {
```
Comment (fgan1) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/sdk/AzureVirtualMachineOperationSDKTest.java]: The method name changed so It's must be changed in the tests context.
findVirtualMachineSizeName -> **findVirtualMachineSize**
Please, review it in whole test class.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java (hunk 3 above)
Comment (wassoares) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java]: Ok, I get it.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java (hunk 4 above)
Comment (wassoares) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java]: Ok, I'll rename.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureResourceIdBuilder.java (hunk 5 above)
Comment (wassoares) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/util/AzureResourceIdBuilder.java]: Ok.

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/sdk/AzureVirtualMachineOperationSDKTest.java (hunk 6 above)
Comment (wassoares) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/sdk/AzureVirtualMachineOperationSDKTest.java]: Sorry, I thought I checked this before, but it looks like I needed to fix this one.

Comment (wassoares) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/sdk/AzureVirtualMachineOperationSDKTest.java]: Ok, I'll make this change now.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java (hunk 1 above)
Comment (wassoares) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureComputePlugin.java]: Ok, is done.

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureVirtualMachineOperation.java (hunk 2 above)
Comment (wassoares) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/AzureVirtualMachineOperation.java]: Ok, is done.

**Pull request merged**